src/Microsoft.AspNet.Razor/RazorError.cs

```diff
@@ -46,19 +61,23 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var err = obj as RazorError;
-            return (err != null) && Equals(err);
+            var error = obj as RazorError;
+            return Equals(error);
```

Comment:
have no `RazorError` subclass scenario; same for every case where `Equals()` does not check for `Type` match.